src/api/v1/public/routes/apps.js

```diff
@@ -181,45 +209,119 @@ app.patch("/", function(req, res)
     validateUpdateAppPayload(req.body)
     .then(function (payload)
     {
-        const response = new SuccessResponse({
-            res,
-            message: `Successfully validated update app payload`,
-        });
-        res.send(response);
-        //res.json(response);
+        const findParams = req.body.old;
+        const updateObj = req.body.new;
 
-        /*
-        AppController.updateOne(req.body)
+        AppController.updateOne(findParams, updateObj)
         .then(function (data)
         {
             const response = new SuccessResponse({
                 res,
-                message: `Successfully created an app named ${req.body.displayName}`,
-                data: data,
+                message: getSuccessMessageForUpdateApps(findParams),
+                data: data.results,
             });
-            res.send(response);
+            res.json(response);
         })
         .catch(function (err)
         {
-            const errResponse = new BadRequestErrorResponse({
-                res,
-                message: `Failed to create an app named ${req.body.displayName}`,
-                err,
-            });
-            res.send(errResponse);
+            // Mongo Error
+            if (err && err.status && err.status === 500)
```

Comment:
This occurs if no value was found that could be updated. We want to distinguish that from a normal error.